threescale_api/client.py

```diff
@@ -247,7 +247,8 @@ def request(self, method='GET', url=None, path='', params: dict = None,
         Returns:
 
         """
-
+        if 'resource' in kwargs:
```

Comment:
this looks alien. if 'resource' has something with 'crd' this should be probably handled elsewhere.

Reply:
`resource` is part of `kwargs` of `delete` function, see `threescale_api/defaults.py` and:
`381     def delete(self, **kwargs):
382         self.client.delete(entity_id=self.entity_id, resource=self, **kwargs)`
This is not a problem for this client, but it creates side effects when I inherit from this `DefaultClient` class.

Reply:
it is also used in
```
376         new_entity = self.client.update(entity_id=self.entity_id,
377                                         params=new_params,
378                                         resource=self,
379                                         **kwargs)
```
I've added `resource` to delete function, because having object id in delete function is enough for threescale client but I need to have resource object in CRD client. The resource can pass itself to the client method.